addons/docs/src/frameworks/angular/sourceDecorator.ts

```diff
@@ -0,0 +1,57 @@
+import { addons, StoryContext } from '@storybook/addons';
+import { computesTemplateSourceFromComponent } from '@storybook/angular/renderer';
+import { logger } from '@storybook/client-logger';
+import prettierHtml from 'prettier/parser-html';
+import prettier from 'prettier/standalone';
+import { SNIPPET_RENDERED, SourceType } from '../../shared';
+
+export const skipSourceRender = (context: StoryContext) => {
+  const sourceParams = context?.parameters.docs?.source;
+
+  // always render if the user forces it
+  if (sourceParams?.type === SourceType.DYNAMIC) {
+    return false;
+  }
+  // never render if the user is forcing the block to render code, or
+  // if the user provides code, or if it's not an args story.
+  return (
+    // (!isArgsStory && context?.parameters) ||
+    sourceParams?.code || sourceParams?.type === SourceType.CODE
+  );
+};
+
+const prettyUp = (source: string) => {
+  return prettier.format(source, {
+    parser: 'angular',
+    plugins: [prettierHtml],
+    htmlWhitespaceSensitivity: 'ignore',
+  });
+};
+
+/**
+ * Svelte source decorator.
+ * @param storyFn Fn
+ * @param context  StoryContext
+ */
+export const sourceDecorator = (storyFn: any, context: StoryContext) => {
```

Comment:
```suggestion
export const sourceDecorator = (storyFn: StoryFn<IStory>, context: StoryContext) => {
```

`import { IStory } from '@storybook/angular';`